File: hacker_rank/PALG-2-simple-array-sum/python/main.py
#!/bin/python3

import sys


def parse_input(text):
    lines = text.split('\n')
    n = int(lines[0])
    numbers = [int(i) for i in lines[1].split()]
    return n, numbers


def main(n, numbers):
    return sum(numbers)


if __name__ == '__main__':
    args = parse_input(sys.stdin.read())
    print(main(*args))

File: hacker_rank/PALG-2-simple-array-sum/rust/src/main.rs
use std::io;
use std::io::prelude::*;
use std::num::ParseIntError;


fn parse_input() -> Result<(i32, Vec<i32>), ParseIntError> {
    let stdin = io::stdin();
    let mut lines: Vec<String> = stdin
        .lock()
        .lines()
        .take(2)
        .filter_map(|line| line.ok())
        .collect();

    let count: i32 = lines.remove(0).parse()?;
    let items: Vec<i32> = lines.remove(0).split_whitespace().map(|i| {
        i.parse().unwrap()
    }).collect();

    Ok((count, items))
}


fn main() {
    match parse_input() {
        Ok((x, y)) => {
            let res: i32 = y.iter().sum();
            println!("{}", res);
        },
        Err(_) => println!("Unable to parse inputs"),
    }
}
